app.py

```diff
@@ -138,6 +138,20 @@ def register_proposal(caller_identity):
     if proposal is None:
         return jsonify(error='missing service_proposal'), 400
 
+    service_type = proposal.get('service_type', None)
+
+    # check if the service type is defined.
+    # If defined, check if it's in the possible enum values
+    if service_type is not None and \
+            service_type not in ServiceType.__members__:
+        return jsonify(
+            error='invalid service_type {}'.format(service_type)
+        ), 400
+
+    # Default to openvpn if service_type is not specified
+    if service_type is None:
+        proposal['service_type'] = ServiceType.openvpn.name
+
     node_key = proposal.get('provider_id', None)
     if node_key is None:
         return jsonify(error='missing provider_id'), 400
```

Comment:
API should validate existence of required field `service_type`, same as `provider_id` here

Reply:
this breaks backwards compatibility. If we don't care about it - sure, otherwise we'll need to move this to a v2 endpoint and do it gradually

Reply:
But all callers are passing `service_type` in payload currently

Reply:
fair point. lol.